src/core/algorithms/statistics/data_stats.cpp

```diff
@@ -505,13 +487,119 @@ Statistic DataStats::GetMedianAD(size_t index) const {
+}
+
+Statistic DataStats::GetNumberOfNonLetterChars(size_t index) const {
+    if (all_stats_[index].num_non_letter_chars.HasValue())
+        return all_stats_[index].num_non_letter_chars;
+
+    auto pred = [](int symbol) { return !std::isalpha(symbol); };
+
+    return CountIfInColumn(pred, index);
+}
+
+Statistic DataStats::GetNumberOfDigitChars(size_t index) const {
+    if (all_stats_[index].num_digit_chars.HasValue()) return all_stats_[index].num_digit_chars;
+
+    auto pred = [](int symbol) { return std::isdigit(symbol); };
+
+    return CountIfInColumn(pred, index);
+}
+
+Statistic DataStats::GetNumberOfLowercaseChars(size_t index) const {
+    if (all_stats_[index].num_lowercase_chars.HasValue())
+        return all_stats_[index].num_lowercase_chars;
+
+    auto pred = [](int symbol) { return std::islower(symbol); };
+
+    return CountIfInColumn(pred, index);
+}
+
+Statistic DataStats::GetNumberOfUppercaseChars(size_t index) const {
+    if (all_stats_[index].num_uppercase_chars.HasValue())
+        return all_stats_[index].num_uppercase_chars;
+
+    auto pred = [](int symbol) { return std::isupper(symbol); };
+
+    return CountIfInColumn(pred, index);
+}
+
+template <class Pred>
+Statistic DataStats::CountIfInColumn(Pred pred, size_t index) const {
     mo::TypedColumnData const& col = col_data_[index];
-    size_t count = col.GetNumNulls();
+    if (col.GetTypeId() != +mo::TypeId::kString) return {};
+
+    size_t count = 0;
+    std::string string_data;
     mo::IntType int_type;
 
-    return Statistic(int_type.MakeValue(count), &int_type, false);
+    for (size_t i = 0; i < col.GetNumRows(); i++) {
+        string_data = col.GetDataAsString(i);
+        for (size_t j = 0; j < string_data.size(); j++)
+            if (pred(string_data[j])) count++;
+    }
+
+    std::byte const* res = int_type.MakeValue(count);
+
+    return Statistic(res, &int_type, false);
+}
+
+Statistic DataStats::GetNumberOfChars(size_t index) const {
+    mo::TypedColumnData const& col = col_data_[index];
+    if (col.GetTypeId() != +mo::TypeId::kString) return {};
+
+    size_t count = 0;
+    std::string string_data;
+    mo::IntType int_type;
+
+    for (size_t i = 0; i < col.GetNumRows(); i++) {
+        string_data = col.GetDataAsString(i);
+        count += string_data.length();
+    }
+
+    std::byte const* res = int_type.MakeValue(count);
+
+    return Statistic(res, &int_type, false);
+}
+
+Statistic DataStats::GetAvgNumberOfChars(size_t index) const {
+    if (all_stats_[index].num_avg_chars.HasValue()) return all_stats_[index].num_avg_chars;
+
+    mo::TypedColumnData const& col = col_data_[index];
+    if (col.GetTypeId() != +mo::TypeId::kString) return {};
+
+    mo::DoubleType double_type;
+    mo::IntType int_type;
+    std::byte* res = double_type.Allocate();
+    Statistic num_stat = GetNumberOfChars(index);
+    std::byte const* num = num_stat.GetData();
+    std::byte const* count = double_type.MakeValueOfInt(col_data_.size());
```

Comment:
```suggestion
    std::byte const* count = double_type.MakeValueOfInt(col.GetNumRows());
```
```suggestion
    std::byte const* count = double_type.MakeValueOfInt(col.GetNumRows() - col.GetNumNulls());
```
Not sure which one is better.